From 1df9a4ec2f4fbdda469765a42d6c3ba14713f9e0 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Fri, 23 Feb 2024 09:14:58 +0100
Subject: [PATCH 0649/1695] dts: stm32: activate HSI calibration on STM32MP25
 boards

Default enable HSI (High Speed Internal oscillator) calibration. This
internal oscillator tends to derive over time so calibration is needed.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e6b3a8b7fb3c953eb410377858d3d1e912616b17 dts: stm32: activate HSI calibration on STM32MP25 boards]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I310188ff86638612e6f4266efd6b73f28d9136d4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/295290
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/310954
Tested-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362469
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp257f-dk.dts  | 4 ++++
 core/arch/arm/dts/stm32mp257f-ev1.dts | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index 741eed10f..f819e8b0f 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -48,6 +48,10 @@
 	};
 };
 
+&hsi_calibration {
+	status = "okay";
+};
+
 &i2c7 {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&i2c7_pins_a>;
diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index 6a9627f2f..6ffc7dfe1 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -48,6 +48,10 @@
 	};
 };
 
+&hsi_calibration {
+	status = "okay";
+};
+
 &i2c7 {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&i2c7_pins_a>;
-- 
2.47.2

